eng/Publishing.props

```diff
@@ -55,11 +75,85 @@
         <PublishFlatContainer>true</PublishFlatContainer>
         <RelativeBlobPath>$(_UploadPathRoot)/$(_PackageVersion)/%(Filename)%(Extension)</RelativeBlobPath>
       </ItemsToPushToBlobFeed>
-      <ItemsToPushToBlobFeed Include="@(_CliFilesToPublish)">
+      <ItemsToPushToBlobFeed Include="@(_CliFileToPublish)">
         <IsShipping>false</IsShipping>
         <PublishFlatContainer>true</PublishFlatContainer>
         <RelativeBlobPath>$(_UploadPathRoot)/$(_PackageVersion)/%(Filename)%(Extension)</RelativeBlobPath>
       </ItemsToPushToBlobFeed>
     </ItemGroup>
   </Target>
+
+  <UsingTask TaskName="CheckExpectedAspireCliArchives" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
```

Comment:
Not a fan of roslyn inline tasks. This is a perf bottleneck and also a complication in the build.  Can we instead just do this in m sbuild logic to make sure the expected files are there?

Reply:
> This is a perf bottleneck

This will run once in the internal build before publishing the archives.

>  Can we instead just do this in m sbuild logic to make sure the expected files are there?

That was my first approach but it gets complicated. This is much cleaner IMHO.

Reply:
We have been victims in the past of the version of roslyn/msbuild that gets used for these types of inline tasks in the past and regreted it. In runtime they have moved off from inline tasks and instead built tasks assemblies due to those issues. I won't block this PR for this but I do think that this is something that should get cleaned up, as it goes very much against the golas we have for simplification of the infrastructure.

Reply:
We can add task assemblies but then we need to add stuff to make sure they get built early, and the fresh build is initiated to allow msbuild to pick up those tasks. Runtime has support for build split up like that. We can explore doing similar work for aspire.

Reply:
I'll do this in msbuild.

Reply:
done